frontend/amundsen_application/static/js/ducks/search/api/v0.ts

```diff
@@ -55,38 +58,29 @@ export const isResourceIndexed = (resource: ResourceType) => {
   return false;
 };
 
-export function searchResource(
+export function search(
   pageIndex: number,
-  resource: ResourceType,
-  term: string,
+  resultsPerPage: number,
+  resources: ResourceType[],
+  searchTerm: string,
   filters: ResourceFilterReducerState = {},
   searchType: SearchType
 ) {
-  /* If resource support is not configured or if there is no search term for non-filter supported resources*/
-  if (
-    resource === ResourceType.user &&
-    (!indexUsersEnabled() || term.length === 0)
-  ) {
-    return Promise.resolve({});
-  }
-  if (!isResourceIndexed(resource)) {
+  // If given invalid resource in list dont search for that one only for valid ones
+  const validResources = resources.filter((r) => isResourceIndexed(r));
+  if (validResources.length < 1) {
```

Comment:
I would usually expect this instead:
```suggestion
  if (!validResources.length) {
```